client-rpc/src/rpc/wallet_rpc.rs

```diff
@@ -132,6 +158,16 @@ where
         Ok(request.name)
     }
 
+    fn restore_watch(&self, request: WalletRequest, view_key: String) -> Result<String> {
```

Comment:
`view_key: SecUtf8`?

Reply:
> One thing to check: are we using the hardened derivation for HD-wallet? otherwise by knowing the view key, one can get the rest

I guess view key should be generated as a leaf, not the root key.

Reply:
> `view_key: SecUtf8`?

Done

Reply:
> It's generated the same as a transfer address

It depends on whether it's hardened: https://bitcoin.stackexchange.com/questions/56916/derivation-of-parent-private-key-from-non-hardened-child

Reply:
> > It's generated the same as a transfer address
> 
> It depends on whether it's hardened: https://bitcoin.stackexchange.com/questions/56916/derivation-of-parent-private-key-from-non-hardened-child

Oh, I see, it's not hardened at the lowest level. Use a different account index for view key would ok then.

Reply:
not sure if this breaks anything -- @leejw51 @leejw51crypto @devashishdxt -- but I guess it'll be good to change, or add a warning fixme note

Reply:
It may break a few test cases. Also, the generated addresses will not be backwards compatible. But, I think we shouldn't worry about backwards compatibility right now because HD wallets were not a part of our last release.

Reply:
I'll implement this in a separate PR then.